### PR TITLE
fix: pre and post localstore initiation migration steps

### PR DESCRIPTION
**Pull request by istae**
### Checklist

- [x] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md).
- [ ] My change requires a documentation update, and I have done it.
- [x] I have added tests to cover my changes.
- [ ] I have filled out the description and linked the related issues.

### Description
Because the retrievalIndexItem's field is changing, we need to make sure all old items are migrated before any other recovery and migration takes place due to unmarshalling errors.

As a result, the migration is now in two parts, pre and post migration. The pre migration will run before any other recovery and migration process of the localstore.

The PR also contains a new field in /debugstore

### CODE REVIEW

File: pkg/storage/migration/migration.go [hunk 1]
```diff
@@ -28,12 +28,12 @@ var (
 )
 
 // Migrate migrates the storage to the latest version
-func Migrate(s storage.BatchedStore, sm Steps) error {
+func Migrate(s storage.BatchedStore, group string, sm Steps) error {
```
Comment (mrekucci) [inline pkg/storage/migration/migration.go:31]: This `group` parameter should be properly documented, e.g.: what effect it has.

File: pkg/storer/migration/all_steps.go [hunk 2]
```diff
@@ -10,8 +10,8 @@ import (
 	"github.com/ethersphere/bee/pkg/storer/internal/reserve"
 )
 
-// AllSteps lists all migration steps for localstore IndexStore.
-func AllSteps(
+// PostSteps lists all migration steps for localstore IndexStore after the localstore is intiated.
```
Comment (mrekucci) [inline pkg/storer/migration/all_steps.go:13]: I'd suggest naming it `before` and `after`.

File: pkg/storer/debug.go [hunk 3]
```diff
@@ -30,10 +31,11 @@ type CacheStat struct {
 }
 
 type ReserveStat struct {
```
Comment (mrekucci) [inline pkg/storer/debug.go:33]: It would be good if this is reflected in the OpenAPI spec.

**Review by mrekucci — APPROVED** [review]
Apart from the comments, LGTM!

File: pkg/storer/debug.go (hunk 3 above)
Comment (istae) [reply pkg/storer/debug.go:33]: we dont have an entry in openAPI for /debugstore

Comment (mrekucci) [reply pkg/storer/debug.go:33]: If it is not a temporary API, it should be documented in the OpenAPI specification.

**Review by mrekucci — APPROVED** [review]

File: pkg/storer/migration/all_steps.go [hunk 4]
```diff
@@ -21,6 +21,12 @@ func AllSteps(
 		2: step_02,
 		3: step_03(chunkStore, reserve.ChunkType),
 		4: step_04(sharkyPath, sharkyNoOfShards),
-		5: step_05,
+	}
+}
+
+// BeforeIinitSteps lists all migration steps for localstore IndexStore before the localstore is intiated.
+func BeforeIinitSteps() migration.Steps {
```
[the remark this answers is not in the document]
Comment (istae) [reply pkg/storer/migration/all_steps.go:28]: The problem with this approach is that it gets messy very quickly. For example, if sharky recovery runs when running the new release, which item do you use?

**Pull request merged**